yodeploy/tests/test_util.py

```diff
@@ -77,18 +78,18 @@ def test_permission_squash(self):
             'PATH': os.environ['PATH'],
             'PYTHONPATH': ':'.join(sys.path),
         }
-        subprocess.check_call((
-            'fakeroot',
-            'python',
-            '-c', (
-                'import yodeploy.util, os; '
-                'yodeploy.util.extract_tar("%s", "%s"); '
-                's = os.stat("%s"); '
-                'assert s.st_uid == 0; '
-                'assert s.st_gid == 0'
-            ) % (self.tmppath('test.tar.gz'), self.tmppath('extracted'),
-                 self.tmppath('extracted/bar'))
-        ), env=env)
+        test_script = (
+            'import yodeploy.util, os; '
+            'yodeploy.util.extract_tar("%s", "%s"); '
+            's = os.stat("%s"); '
+            'assert s.st_uid == 0; '
+            'assert s.st_gid == 0'
+        ) % (
+            self.tmppath('test.tar.gz'), self.tmppath('extracted'),
+            self.tmppath('extracted/bar'))
+
+        subprocess.check_call(
+            ('fakeroot', sys.executable, '-c', "'%s'" % test_script), env=env)
```

Comment:
I don't think I said yesterday, but thanks for digging into it with me.

There's a number of places in the test suite where we're writing static python to temp files and then executing those scripts. I didn't know if that was a consequence of preference or convenience. I'd like to pull any such instances out into static files (against master) if you don't have any objections.

https://github.com/yola/yodeploy/blob/998f3326c4426c70cfb017629df8eaf771e90bfb/yodeploy/tests/test_ipc_logging.py#L83-L96

There's also this stuff, which I'd probably leave as is for now:

https://github.com/yola/yodeploy/blob/998f3326c4426c70cfb017629df8eaf771e90bfb/yodeploy/tests/test_application.py#L175-L186

My preference is that there's a fixtures directory with static variants of these (more like the integration_tests are setup), but for whatever reason my brain didn't have as hard a time processing those instances.